source/fundamentals/connection/connect.txt

```diff
@@ -61,146 +76,140 @@ a different dbpath, but all using the same replica set name "foo".
+   set using background threads. If you create a client and immediately
+   print the string representation of its
+   ``~pymongo.mongo_client.MongoClient.nodes`` attribute, the list may be
+   empty initially. Add a pause to your code to allow MongoClient to discover the whole replica
+   set:
 
-.. code-block:: python
+   .. code-block:: python
 
-  >>> from time import sleep
-  >>> c = MongoClient(replicaset='foo'); print(c.nodes); sleep(0.1); print(c.nodes)
-  frozenset([])
-  frozenset([('localhost', 27019), ('localhost', 27017), ('localhost', 27018)])
+      >>> from time import sleep
+      >>> c = MongoClient(replicaset='foo'); print(c.nodes); sleep(0.1); print(c.nodes)
+      frozenset([])
+      frozenset([('localhost', 27019), ('localhost', 27017), ('localhost', 27018)])
 
-You need not wait for replica set discovery in your application, however.
-If you need to do any operation with a MongoClient, such as a
-the ``~pymongo.collection.Collection.find`` method or an
-the ``~pymongo.collection.Collection.insert_one`` method, the client waits to discover
-a suitable member before it attempts the operation.
+     You do not need to wait for replica set discovery for your application to
+     perform operations. The client waits to discover a suitable member before it
+     attempts any operation.
 
 Handling Failover
 ~~~~~~~~~~~~~~~~~
 
-When a failover occurs, PyMongo will automatically attempt to find the
-new primary node and perform subsequent operations on that node. This
-can't happen completely transparently, however. Here we'll perform an
-example failover to illustrate how everything behaves. First, we'll
-connect to the replica set and perform a couple of basic operations:
+When a failover occurs, PyMongo automatically attempts to find the
+new primary node and perform subsequent operations on that node.
+This example shows how a failover behaves. 
+
+Connect to the replica set and perform some basic operations:
 
 .. code-block:: python
 
-  >>> db = MongoClient("localhost", replicaSet='foo').test
-  >>> db.test.insert_one({"x": 1}).inserted_id
-  ObjectId('...')
-  >>> db.test.find_one()
-  {'x': 1, '_id': ObjectId('...')}
+   >>> db = MongoClient("localhost", replicaSet='foo').test
+   >>> db.test.insert_one({"x": 1}).inserted_id
+   ObjectId('...')
+   >>> db.test.find_one()
+   {'x': 1, '_id': ObjectId('...')}
 
-By checking the host and port, we can see that we're connected to
-*localhost:27017*, which is the current primary:
+By checking the host and port, you can see that you're connected to the primary member,
+``localhost:27017``:
 
 .. code-block:: python
 
-  >>> db.client.address
-  ('localhost', 27017)
+   >>> db.client.address
+   ('localhost', 27017)
 
-Now let's bring down that node and see what happens when we run our
-query again:
+Take the node offline and attempt another operation:
+.. I think there's a line of code missing - how are we taking the node offline?
```

Comment:
Good point. The original docs don't say how they brought down the server. 

Based on [this server doc](https://www.mongodb.com/docs/manual/tutorial/remove-replica-set-member/#remove-members-from-replica-set), I _think_ it should read like this:

```
mongosh
db.shutdownServer()
```

For me, that produced: `MongoNetworkError: connection 3 to 127.0.0.1:27017 closed`

When I tried to continue performing Python operations in an existing session, they still worked; MongoClient immediately reconneded to localhost:27018. When I exited and restarted my Python session, then tried to connect to localhost:27017, I got this:

```
raise ServerSelectionTimeoutError(
pymongo.errors.ServerSelectionTimeoutError: localhost:27017: [Errno 61] Connection refused, Timeout: 30s, Topology Description: <TopologyDescription id: 65d75c354fa8fe42017e0c34, topology_type: Unknown, servers: [<ServerDescription ('localhost', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('localhost:27017: [Errno 61] Connection refused')>]>
```

It's possible I just wasn't fast enough to test the find operation after bringing down the primary node. The results could be different in an application with split-second timing. For this page, we could describe what I experienced above (MongoDB elects a new primary and the driver fails over to it, so the query still works), but mention that there might be a short period during failover when the query wouldn't work. What do you think?

Reply:
Based on that doc, I think the correct command is ``rs.remove("localhost:27017")``
I think proper protocol is to shut down the server before removing the member, but in this case, we're trying to replicate an error state, not manage our database.